mirage/factories/comment-report.ts

```diff
@@ -0,0 +1,27 @@
+import { association, Factory, faker, ModelInstance } from 'ember-cli-mirage';
+
+import CommentReport from 'ember-osf-web/models/comment-report';
+import { guid } from './utils';
+
+export default Factory.extend<CommentReport>({
+    id: guid('comment-report'),
+    afterCreate(commentReport: ModelInstance<CommentReport>, server: any) {
+        const reporter = commentReport.comment.user.id;
+        commentReport.update({
+            reporter,
+        });
+
+        const comment = server.schema.comments.find(commentReport.comment.id);
```

Comment:
It's confusing having both ember models and mirage models... since this in mirage, I think `commentReport` and `commentReport.comment` are both mirage `ModelInstance`s. If this were in app code and `commentReport` came from the store, it would be an ember model and `commentReport.comment` would be a promise object.